### PR TITLE
refactor: append group option

### PR DESCRIPTION
**Pull request by oceanroleplay**
```ts
@Discord()
@SlashGroup({ name: "testx" })
@SlashGroup({ appendToChild: true, name: "add", root: "testx" })
export abstract class Group {
  @Slash()
  x(interaction: CommandInteraction): void {
    interaction.reply(":wave:");
  }

  @Slash()
  y(interaction: CommandInteraction): void {
    interaction.reply(":wave:");
  }
}

@Discord()
@SlashGroup({ appendToChild: true, name: "testx" })
export abstract class AnotherGroup {
  @Slash()
  m(interaction: CommandInteraction): void {
    interaction.reply(":wave:");
  }

  @Slash()
  n(interaction: CommandInteraction): void {
    interaction.reply(":wave:");
  }
}
```

### CODE REVIEW

File: packages/discordx/src/decorators/decorators/SlashGroup.ts [hunk 1]
```diff
@@ -30,20 +36,28 @@ export function SlashGroup(info: SlashGroupParams): ClassMethodDecorator {
   ) {
     const myClass = target as unknown as new () => unknown;
 
-    if (descriptor) {
+    if (descriptor || info.appendToChild) {
       // If @SlashGroup decorate a method edit the method and add it to subgroup
       MetadataStorage.instance.addModifier(
-        Modifier.create<DApplicationCommand>((original) => {
-          if (original.type === "CHAT_INPUT") {
```
Comment (AndyClausen) [inline packages/discordx/src/decorators/decorators/SlashGroup.ts:37]: Was this condition meant to be removed?

File: packages/discordx/src/decorators/decorators/Bot.ts [hunk 2]
```diff
@@ -13,20 +13,11 @@ import {
 
 /**
  * Execute your application command, button, select menu, simple command or event by defined bot when multiple bots are running in the same instance
- * @param botId id of your bot
```
Comment (AndyClausen) [inline packages/discordx/src/decorators/decorators/Bot.ts:16]: Why did you remove all the param descriptions?

Comment (oceanroleplay) [reply packages/discordx/src/decorators/decorators/Bot.ts:16]: image

When we use our IDE, the options are already there, so it feel like duplicate descriptions.

**Comment by VictoriqueMoe** [comment]
Why did you merge this PR before the comments where raised and addressed? @oceanroleplay ?

**Comment by oceanroleplay** [comment]
> Why did you merge this PR before the comments where raised and addressed? @oceanroleplay ?

comments are after, not before.

**Comment by VictoriqueMoe** [comment]
> 

No, PR review comments are for BEFORE you merge a PR into dev.

Raise PR
Assign Reviewers
Address comments and issues
Reviews approve
Merge PR

This is how it works

**Comment by oceanroleplay** [comment]
maybe bad move, I will create new comments

**Comment by oceanroleplay** [comment]
> > 
> 
> No, PR review comments are for BEFORE you merge a PR into dev.
> 
> Raise PR Assign Reviewers Address comments and issues Reviews approve Merge PR
> 
> This is how it works

it's after 
image

**Comment by VictoriqueMoe** [comment]
yes, never merge a PR until it has passed review, or else how do you revert a portentously broken change? 

The whole reason you code on a new branch is for protection

**Comment by VictoriqueMoe** [comment]
> > > 
> > 
> > 
> > No, PR review comments are for BEFORE you merge a PR into dev.
> > Raise PR Assign Reviewers Address comments and issues Reviews approve Merge PR
> > This is how it works
> 
> it's after image

I know it was done after. i said this is wrong and you should have never merged it.

**Comment by oceanroleplay** [comment]
This is already in todo list of improvement https://github.com/oceanroleplay/discord.ts/discussions/524

The comments are removed in this PR, but they will be back with improved comments. That's been already decided before merging this PR.

File: packages/discordx/src/decorators/decorators/Bot.ts (hunk 2 above)
Comment (oceanroleplay) [reply packages/discordx/src/decorators/decorators/Bot.ts:16]: They will be back with improved version
